### PR TITLE
Refactor code, add Redux

### PR DESCRIPTION
**Pull request by offcast-dev**
PT: https://www.pivotaltracker.com/story/show/178336690
Code can have some refactoring
### Changes Proposed
- extract axios  links to variable 
- refactor Navigator option into screenOptions
- refactors conditions and ternaries
- adds Redux
- refactors code to save responses in store
### What Learned
- using Redux in React Native

### CODE REVIEW

**Review by lukeperrin10 — APPROVED** [review]
Nice refactor, App jsx is a thing of beauty now :+1:

File: modules/Articles.js [hunk 1]
```diff
@@ -1,23 +1,35 @@
 import axios from 'axios';
+import store from '../state/store/configureStore';
 
+const uri = 'https://fakest-newzz.herokuapp.com/api/articles';
```
Comment (LucasKnudsen) [inline modules/Articles.js:4]: axiosBaseUrl no?

File: views/MainView.jsx [hunk 2]
```diff
@@ -1,30 +1,27 @@
...
-  const fetchArticles = async () => {
-    const response = await Articles.getAll();
-    response[0] ? setArticles(response) : setNoArticlesMessage(true);
-  };
+  const { articles } = useSelector((state) => state);
 
   useEffect(() => {
-    fetchArticles();
+    Articles.getAll();
   }, []);
```
Comment (LucasKnudsen) [inline views/MainView.jsx:13]: Articles.index like the other ones ? :3

File: views/SingleArticleView.jsx [hunk 3]
```diff
@@ -8,32 +8,27 @@ import {
...
-      category: singleArticle.category,
+    props.navigation.navigate('single category view', {
+      category: article.category,
     });
   };
 
   useEffect(() => {
-    fetchArticle();
+    Articles.getSpecific(articleId);
   }, []);
```
Comment (LucasKnudsen) [inline views/SingleArticleView.jsx:25]: show? :))

**Review by LucasKnudsen — APPROVED** [review]

**Pull request merged**